frontend/web/components/pages/EnvironmentSettingsPage.tsx

```diff
@@ -0,0 +1,891 @@
+                              className='no-pad'
+                              title={
+                                <Tooltip
+                                  title={
+                                    <h5 className='mb-0'>
+                                      Webhooks <Icon name='info-outlined' />
+                                    </h5>
+                                  }
+                                  place='right'
+                                >
+                                  {Constants.strings.WEBHOOKS_DESCRIPTION}
+                                </Tooltip>
+                              }
+                            >
+                              <div className='search-list'>
+                                <Row className='list-item p-3 text-muted'>
+                                  You currently have no Feature Webhooks
+                                  configured for this Environment.
+                                </Row>
+                              </div>
+                            </Panel>
+                          }
+                          isLoading={webhooksLoading}
+                        />
+                      )}
+                    </FormGroup>
+                  </TabItem>
+                  {metadataEnable &&
+                    environmentContentType?.id && (
+                      <TabItem tabLabel='Custom Fields'>
+                        <FormGroup className='mt-5 setting'>
+                          <InputGroup
+                            title={'Custom fields'}
+                            tooltip={`${Constants.strings.TOOLTIP_METADATA_DESCRIPTION(
+                              'environments',
+                            )}`}
+                            tooltipPlace='right'
+                            component={
+                              <AddMetadataToEntity
+                                organisationId={
+                                  AccountStore.getOrganisation().id
+                                }
+                                projectId={match.params.projectId}
+                                entityId={currentEnv?.api_key || ''}
+                                envName={currentEnv?.name}
+                                entityContentType={
+                                  environmentContentType?.id
+                                }
+                                entity={
+                                  environmentContentType.model
+                                }
+                              />
+                            }
+                          />
+                        </FormGroup>
+                      </TabItem>
+                    )}
+                </Tabs>
+              )}
+            </>
+          )
+        }}
+      </ProjectProvider>
+    </div>
+  )
+}
+
+
+
+EnvironmentSettingsPageTS.displayName = 'EnvironmentSettingsPage'
+
+export default ConfigProvider(withWebhooks(EnvironmentSettingsPageTS))
```

Comment:
Same here, I do think `withWebhooks` could be turned into a hook so there is no need to pass the props from above.

The methods are not dynamic so I believe it's a good use case to turn it into a hook and avoid an additional wrapping of child component

Reply:
Yeah actually this is probably the easiest migration to RTK I can think of @tiagoapolo

Reply:
Ok nice, if ever you remember, can you ping me on a PR once you start the migration to RTK?

Reply:
@Zaimwa9 this pull request implements useWebhooks, let me know if you have questions.
https://github.com/Flagsmith/flagsmith/pull/5087